_posts/en/newsletters/2021-06-23-newsletter.md

```diff
@@ -0,0 +1,161 @@
+[Rust-Lightning][rust-lightning repo], [libsecp256k1][libsecp256k1
+repo], [Hardware Wallet Interface (HWI)][hwi repo],
+[Rust Bitcoin][rust bitcoin repo], [BTCPay Server][btcpay server repo],
+[Bitcoin Improvement Proposals (BIPs)][bips repo], and [Lightning
+BOLTs][bolts repo].*
+
+- [Bitcoin Core #21365][] adds the ability for the wallet to create
+  signatures for [taproot][topic taproot] spends---both keypath spends
+  using only the P2TR public key and scriptpath spends using a
+  [tapscript][topic tapscript].  The wallet can also sign for
+  taproot-spending [PSBTs][topic psbt], but only if the wallet already
+  has all the keypath or scriptpath information it needs.  The somewhat
+  related merged PR [#22156][bitcoin core #22156] only allows importing
+  that keypath and scriptpath information after taproot is active (block
+  709,632 on mainnet, but on test networks where taproot is already
+  enabled, importing may be used now).
+
+- [Bitcoin Core #22144][] randomizes the order in which peers are serviced
+  in the message handling thread. Bitcoin Core's message handling thread is
+  responsible for parsing and processing p2p messages from peers and for
+  sending messages to those peers. Previously, the message handling thread
+  would service each peer round-robin _in the order in which the
+  connections to those peers were first established_. This PR changes the
+  logic so that, on each iteration of the message handling loop, the order in
+  which peers are serviced is randomized. Peers are still serviced with the same
+  frequency (each peer is serviced once per iteration), but any weaknesses or
+  exploits that rely on a deterministic ordering of servicing peers are
+  avoided.
+
+- [Bitcoin Core #21261][] makes it easier to extend inbound connection
+  diversity protection to more networks and then uses that framework to
+  add [I2P][] the list of diversity-protected networks.  Diversity
+  protection (often called eviction protection) allows a few peers with
+  desirable characteristics to remain connected when Bitcoin Core is
+  otherwise pruning low-quality connections.  Retaining a few
+  connections to peers on anonymity networks is highly desirable both
+  because it allows transaction creators to use those networks to hide
+  their network identity and because the ability to receive blocks over
+  those networks in addition to the regular Internet Protocol can
+  prevent some types of [eclipse attacks][topic eclipse attacks].
+
+- [Rust Bitcoin #601][] adds support for parsing [bech32m][topic bech32]
+  addresses and requires that v1+ native segwit addresses be encoded with
+  bech32m and not bech32.
+
+- [BTCPay Server #2450][] makes generating [payjoin][topic
+  payjoin]-compatible invoices the default when the user opts into using
+  a hot wallet for receiving payments.  A button on the *create wallet*
+  screen allows the user to opt-out of payjoin receiving support.
```

Comment:
(a) no hyphen for the verb form, (b) not sure if this is better but "payjoin receiving support" seems incorrect or awkward
```suggestion
  screen allows the user to opt out of this default setting.
```

Reply:
Blarg, sorry I suck at hyphen usage and you keep needing to correct me.  :-(